src/Symfony/Component/Mime/Address.php

```diff
@@ -77,7 +77,19 @@ public function getEncodedAddress(): string
 
     public function toString(): string
     {
-        return ($n = $this->getName()) ? $n.' <'.$this->getEncodedAddress().'>' : $this->getEncodedAddress();
+        return ($n = $this->getEncodedName()) ? $n.' <'.$this->getEncodedAddress().'>' : $this->getEncodedAddress();
+    }
+
+    public function getEncodedName(): string
+    {
+        $name = $this->getName();
+
+        if (preg_match('/,/u', $name)) {
+            $name = preg_replace('/"/u', '\"', $name);
+            $name = sprintf('"%s"', $name);
+        }
```

Comment:
Is it possible to detect an already encoded name to avoid double escaping?

Reply:
You mean that the developer will instanciate the Address like this:
```php
new Address('fabien@symfony.com', '"Fabien, Potencier"');
```

Reply:
Yes, but is this something wanted?

Reply:
Wouldn't it be easier and (in general terms) better to always quote the name?

Reply:
This would be better if the name is quoted every time because, an issue may occur with some other cases, like @ < >...
Or is there any restriction with quoting ?

Reply:
I will change it to make the encoded address always quoted.

Reply:
Done